rpaas/nginx/configuration_render.go

```diff
@@ -0,0 +1,162 @@
+package nginx
+
+import (
+	"bytes"
+	"text/template"
+
+	"github.com/tsuru/rpaas-operator/pkg/apis/extensions/v1alpha1"
+)
+
+type ConfigurationRenderer interface {
+	Render(ConfigurationData) (string, error)
+}
+
+type ConfigurationData struct {
+	Config   *v1alpha1.NginxConfig
+	Instance *v1alpha1.RpaasInstanceSpec
+}
+
+type rpaasConfigurationRenderer struct {
+	t *template.Template
+}
+
+func (r *rpaasConfigurationRenderer) Render(c ConfigurationData) (string, error) {
+	buffer := &bytes.Buffer{}
+	err := r.t.Execute(buffer, c)
+	return buffer.String(), err
+}
+
+func NewRpaasConfigurationRenderer() ConfigurationRenderer {
+	return &rpaasConfigurationRenderer{
+		t: template.Must(template.New("rpaas-configuration-template").Parse(rawNginxConfiguration)),
```

Comment:
I think it would be okay to move this template to a global variable so that we only have to parse the default template once.

Reply:
Of course...